design/hook-execution.md

```diff
@@ -0,0 +1,68 @@
+# Hook Execution
```

Comment:
This entire document assumes we have a list of different applications for hooks that users requested for,  and we prefer to let them do that themselves with webhooks or local execution. Is that the case (i.e. do we have numerous use-cases, or are we're trying to generalise from the start) ?

Reply:
As I understand the use case is the entry level work needed to use the hooks for integration and by enabling a simple execution mechanism will reduce this entry barrier.

Reply:
Not sure I understand. This more of a product question. I know of a single user which consider webhooks as a struggle. I also think we can manage without running custom code. Are there any other feedbacks supporting that?

Reply:
Quoted the issue by product.
Will take the discussion with product to sharpen the requirements in order to have better understanding the use-case.